consultation_analyser/consultations/models.py

```diff
@@ -40,301 +41,237 @@ class Meta:
         ordering = ["created_at"]
 
 
-class Consultation(UUIDPrimaryKeyModel, TimeStampedModel):
-    name = models.CharField(max_length=256)
-    slug = models.CharField(null=False, max_length=256)
-    users = models.ManyToManyField(User)
-
-    def has_processing_run(self):
-        return ProcessingRun.objects.filter(consultation=self).exists()
+class SlugFromTextModel(models.Model):
+    text = models.CharField(max_length=256)
+    slug = models.SlugField(null=False, editable=False, max_length=256)
 
-    class Meta(UUIDPrimaryKeyModel.Meta, TimeStampedModel.Meta):
-        constraints = [
-            models.UniqueConstraint(fields=["slug"], name="unique_consultation_slug"),
-        ]
+    def save(self, *args, **kwargs):
+        # Generate a slug from the text - ensure unique by adding timestamp if needed.
+        # Don't allow empty slug.
+        ModelClass = self.__class__
+        cropped_length = 220
+        cropped_text = self.text[:cropped_length]
+        generated_slug = slugify(cropped_text)
+        if self.pk:
+            slug_exists = (
+                ModelClass.objects.filter(slug=generated_slug).exclude(pk=self.pk).exists()
+            )
+        else:
+            slug_exists = ModelClass.objects.filter(slug=generated_slug).exists()
+        timestamp = datetime.datetime.now().strftime("%Y%m%d-%H%M%S%f")
+        if not generated_slug:
+            generated_slug = timestamp
+        elif slug_exists:
+            generated_slug = f"{generated_slug}-{timestamp}"
```

Comment:
what do we need from timestamps in slugs?

Reply:
The timestamp is for uniqueness, on the off chance that two questions have the same slug (unlikely in general, but as I found out last time, will def happen a lot in testing when we're uploading the same consultation loads of times).

Reply:
I've also just realised the question slug only needs to be unique up to consultation, not unique overall.